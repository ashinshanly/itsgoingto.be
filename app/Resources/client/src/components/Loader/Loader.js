import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { isLoadingSelector } from 'store/loader'
import './Loader.scss'

export const Loader = ({ isLoading }) => (
  <div className={'loader-container hideable' + (isLoading ? '' : ' gone')}>
    <div className='loader'>
      <p className='loader__label'>?</p>
      <div className='loader__figure' />
    </div>
  </div>
)

Loader.propTypes = {
  isLoading : PropTypes.bool.isRequired,
}

const mapStateToProps = (state) => ({
  isLoading : isLoadingSelector(state),
})

export default connect(mapStateToProps)(Loader)
