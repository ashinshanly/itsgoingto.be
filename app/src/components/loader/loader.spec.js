/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'
import Loader from './loader'

describe('(Component) Loader', () => {
  describe('(Render)', () => {
    describe('when prop isLoading is false', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Loader isLoading={false} />)

        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when prop isLoading is true', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Loader isLoading={true} />)

        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})