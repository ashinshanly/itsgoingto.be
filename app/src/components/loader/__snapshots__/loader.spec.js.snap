// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Loader (Render) when prop isLoading is false matches snapshot 1`] = `
<div
  className="loader-container hideable gone"
>
  <div
    className="loader"
  >
    <p
      className="loader__label"
    >
      ?
    </p>
    <div
      className="loader__figure"
    />
  </div>
</div>
`;

exports[`(Component) Loader (Render) when prop isLoading is true matches snapshot 1`] = `
<div
  className="loader-container hideable"
>
  <div
    className="loader"
  >
    <p
      className="loader__label"
    >
      ?
    </p>
    <div
      className="loader__figure"
    />
  </div>
</div>
`;
